cluster/src/dunit/scala/org/apache/spark/sql/CachedBatchScanDUnitTest.scala

```diff
@@ -0,0 +1,124 @@
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package org.apache.spark.sql
+
+import io.snappydata.cluster.ClusterManagerTestBase
+
+import org.apache.spark.sql.execution.columnar.ColumnTableScan
+import org.apache.spark.sql.execution.metric.SQLMetrics
+
+
+class CachedBatchScanDUnitTest(s: String) extends ClusterManagerTestBase(s){
+
+  def testCachedBatchSkipping(): Unit = {
+
+    val snc = SnappyContext(sc)
+    val ddlStr = "YearI INT NOT NULL," +
+        "MonthI INT NOT NULL," +
+        "DayOfMonth INT NOT NULL," +
+        "DepTime INT," +
+        "ArrDelay INT," +
+        "UniqueCarrier CHAR(6) NOT NULL"
+
+    snc.sql(s"create table if not exists airline ($ddlStr) " +
+          s" using column options (Buckets '2')").collect()
+
+    for (i <- 1 to 100 ) {
+      snc.sql(s"insert into airline values(2015, 2, 15, 1002, $i, 'AA')")
+    }
+
+    // ***Check for the case when all the cached batches are scanned ****
+    var previousExecutionIds = snc.sharedState.listener.executionIdToData.keySet
+
+    var df_allCachedBatchesScan = snc.sql(
+      "select AVG(ArrDelay) arrivalDelay, UniqueCarrier carrier " +
+          "from AIRLINE where  ArrDelay < 101 " +
+          "group by UniqueCarrier order by arrivalDelay")
+
+    df_allCachedBatchesScan.count
+
+    var executionIds =
+      snc.sharedState.listener.executionIdToData.keySet.diff(previousExecutionIds)
+
+    var executionId = executionIds.head
+
+    val (scanned1, skipped1) =
+      findCachedBatchStats(df_allCachedBatchesScan, snc.snappySession, executionId)
+    assert(skipped1 == 0, "All Cached batches should have been scanned")
+    assert(scanned1 > 0, "All Cached batches should have been scanned")
+
+    // ***Check for the case when all the cached batches are skipped****
+    previousExecutionIds = snc.sharedState.listener.executionIdToData.keySet
+
+    val df_noCachedBatchesScan = snc.sql(
+      "select AVG(ArrDelay) arrivalDelay, UniqueCarrier carrier " +
+          "from AIRLINE where ArrDelay > 101  " +
+          "group by UniqueCarrier order by arrivalDelay")
+
+    df_noCachedBatchesScan.count
+
+    executionIds =
+        snc.sharedState.listener.executionIdToData.keySet.diff(previousExecutionIds)
+
+    executionId = executionIds.head
+
+    val (scanned2, skipped2) =
+      findCachedBatchStats(df_allCachedBatchesScan, snc.snappySession, executionId)
+    assert(scanned2 == skipped2, "No Cached batches should have been scanned")
+    assert(skipped2 > 0, "No Cached batches should have been scanned")
+
+    // ***Check for the case when some of the cached batches are scanned ****
+    previousExecutionIds = snc.sharedState.listener.executionIdToData.keySet
+
+    val df_someCachedBatchesScan = snc.sql(
+      "select AVG(ArrDelay) arrivalDelay, UniqueCarrier carrier " +
+          "from AIRLINE where ArrDelay < 20  " +
+          "group by UniqueCarrier order by arrivalDelay")
+
```

Comment:
This test adds checks for a few operators. Do you want to add for other operators too? Perhaps we can take from InMemory test suites which would have better coverage?